svd-rs/src/peripheralinfo.rs

```diff
@@ -0,0 +1,304 @@
+use super::{
+    register::{RegIter, RegIterMut},
+    AddressBlock, BuildError, EmptyToNone, Interrupt, RegisterCluster, RegisterProperties,
+    SvdError, ValidateLevel,
+};
+
+/// Errors from [Peripheral::validate]
+#[derive(Clone, Debug, PartialEq, Eq, thiserror::Error)]
+pub enum Error {
+    /// The peripheral has no registers, but specified a `<registers>` tag.
+    #[error("Peripheral have `registers` tag, but it is empty")]
+    EmptyRegisters,
+}
+
+/// A description of a peripheral in the [device](crate::Device), describing, for example, the [memory mappings](crate::RegisterInfo).
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+#[cfg_attr(feature = "serde", serde(rename_all = "camelCase"))]
```

Comment:
this could be
```suggestion
#[cfg_attr(
        feature = "serde",
        derive(serde::Deserialize, serde::Serialize),
        serde(rename_all = "camelCase")
)]
```
but I have no real opinion on it

Reply:
I'd change this in another PR.